applications/dirgraph_visualization/dirgraph_main.py

```diff
@@ -0,0 +1,637 @@
+import sys
+import pysvzerod
+import pandas as pd
+import matplotlib.pyplot as plt
+import networkx as nx
+import json
+import dash
+from dash import html, dcc
+from dash.dependencies import Input, Output, State
+import plotly.graph_objects as go
+import os
+from dirgraph_utils import set_up_0d_network
+
+
+'''
+This file enables the visualization of 0D simulation results in a web app that displays the 0D network 
+as a directed graph. Users can interactively select nodes to view their parameters and simulation results.
+
+Enter the filepath for your simulation input JSON file and the directory where 
+you want to save the output directed graph as command line arguments to run the script. 
+If you want to save the raw svZeroDSolver simulation results, add "export-csv" as the third command line argument.
+
+'''
+
+def dirgraph(filepath, output_dir, export_csv):
+    solver = pysvzerod.Solver(filepath)
+    solver.run()
+    results = pd.DataFrame(solver.get_full_result())
+
+    if export_csv:
+        results.to_csv('results.csv', sep=',', index=False, encoding='utf-8')
+        print(f"Results exported to results.csv")
+
+    with open(filepath, 'r') as infile:
+        parameters = json.load(infile)
+
+    set_up_0d_network(
+        filepath,
+        name_type='id', # Options 'name' or 'id' specifies whether vessel names or ids should be used for each node
+        draw_directed_graph= False,  # Enter True if you want to save the directed graph
+        output_dir= output_dir
+    )
+    base_name = filepath.rsplit('/', 1)[-1]
+    output_file = os.path.join(output_dir, os.path.splitext(base_name)[0] + "_directed_graph.dot")
+    G = nx.DiGraph(nx.nx_pydot.read_dot(output_file))
+    return results, parameters, G
+
+
+def main():
+    # Check if the correct number of arguments is provided
+    if len(sys.argv) < 3:
```

Comment:
Would be good to have `argparse` providing a help message when the user types `python dirgraph_main.py --help`. Instructions are [here.](https://docs.python.org/3/library/argparse.html#help) 

Might also be more intuitive if the main script is called `visualize_simulation.py` or something, instead of `digraph...`